File: 600/solution.py
def cal2(a):
    l = a % 2 + 1
    r = a + 1
    le = (r - l) / 2 + 1
    return (l + r) * le / 2


def cal(a):
    ret = 0
    for i in range(1, a):
        if i * 3 > a:
            break
        ret += cal2(a - i * 3)
    return ret


upper = 55106
ans = 0
for i in range(0, upper):
    if i * 3 + 6 > upper:
        break
    ans += cal(int((upper - i * 3) / 2))
print(ans)

File: 600/solution.cpp
#include <bits/stdc++.h>
using namespace std;
long long cal2(int x) {
    auto lower = x % 2 + 1;
    auto upper = x + 1;
    auto len   = (upper - lower) / 2 + 1;
    return 1ll * (lower + upper) * len / 2;
}
long long cal(int x) {
    long long ret = 0;
    for (int a = 1; a * 3 <= x; a++)
        ret += cal2(x - a * 3);
    return ret;
}
int upper = 55106;
int main() {
    long long ans = 0;
    for (int d = 0; d * 3 + 6 <= upper; d++)
        ans += cal((upper - d * 3) / 2);
    cout << ans << endl;
    return 0;
}
